internal/consensus/pbts_test.go

```diff
@@ -182,51 +207,97 @@ func (p *pbtsTestHarness) nextHeight(proposer types.PrivValidator, deliverTime,
 	signAddVotes(p.ctx, p.observedState, tmproto.PrecommitType, p.chainID, bid, p.otherValidators...)
 	ensurePrecommit(p.t, p.ensureVoteCh, p.currentHeight, p.currentRound)
 
+	vk, err := p.observedValidator.GetPubKey(context.Background())
+	require.NoError(p.t, err)
+	res := collectHeightResults(p.ctx, p.t, p.eventCh, p.currentHeight, vk.Address())
+	ensureNewBlock(p.t, p.blockCh, p.currentHeight)
+
 	p.currentHeight++
 	incrementHeight(p.otherValidators...)
-	return <-p.resultCh
+	return res
 }
 
-// nolint: lll
-func registerResultCollector(ctx context.Context, t *testing.T, eb *eventbus.EventBus, address []byte) <-chan heightResult {
+func timestampedCollector(ctx context.Context, t *testing.T, eb *eventbus.EventBus) <-chan timestampedEvent {
 	t.Helper()
-	resultCh := make(chan heightResult, 2)
-	var res heightResult
+	eventCh := make(chan timestampedEvent, 100)
```

Comment:
I'm not sure why the buffer here needs to be this long—is this basically just "infinitely long"? Does the system deadlock if it had to rendezvous?

Reply:
Yeah, the system does deadlock if this is not long enough. The test uses the synchronous `Observe` method and writes all of the events it sees to this new `eventCh`, annotated with a timestamp. We don't read off of the `eventCh` until the end of each height, so we need the `eventCh` to be large enough to contain all of the messages received each height, otherwise the `Observe` call will block on writing to the channel.

Reply:
Added a comment to clarify.